server/utils/firebase_algolia_server_code/functions/test/index.test.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Description: Unit tests for index.js
+ *              Uses dependency injection to mock algolia save/delete functions
+ *              Uses sinon/chai to spy on calls and validate behavior
+ *              compile with `npm run test`
+ * Author: Robert Marcus
+ * Date: July 7, 2020
+ */
+
+// Import the testing environment configuration
+const { test, assert, sinon } = require('./testConfig');
+
+// Import the generic cloud functions.
+const { createRecordFromEntity, updateRecord, deleteRecord } = 
+    require('../index.js');
+
+// Import mock saveObject, mock deleteObject, and helpers.
+const { saveObject, deleteObject } = require('./algoliaMocks');
+const { getExpectedOutputID, getExpectedOutputSnap, getExpectedOutputChange } =
+    require('./expectedMockOutputHelpers');
+
+describe("test_createRecordFromEntity", () => {
+  it('should call saveObject exactly once', () => {
+    const saveSpy = sinon.spy(saveObject);
+    const snap = test.firestore.exampleDocumentSnapshot();
+
+    const createRecordWrapper = test.wrap(createRecordFromEntity(saveSpy));
+    const addedSnap = createRecordWrapper(snap);
+
+    assert.isTrue(saveSpy.calledOnce);
+  });
+
+  it('should call saveObject with the correct parameters', () => {
+    const snap = test.firestore.exampleDocumentSnapshot();
+    const saveSpy = sinon.spy(saveObject);
+
+    const createRecordWrapper = test.wrap(createRecordFromEntity(saveSpy));
+    const addedSnap = createRecordWrapper(snap);
+
+    assert.isTrue(saveSpy.calledWith(getExpectedOutputSnap(snap)));
```

Comment:
It's generally better to explicitly define the expected output here in the test, so I don't go to another file to understand the test. Even if that means flouting the DRY rule, clarity is more important here.

`const expectedInput = {data: xxx, objectID: yyy};`
`assert spy.calledWith(expectedInput);`

Reply:
+1 I have to keep jumping from file to file to understand what's going on in these tests.  If you can't make it explicit from the code, make it explicit in the comments.

Reply:
Moot, old file.